### PR TITLE
add xnd ruby post

### PR DESCRIPTION
**Pull request by v0dro**
Post with a brief introduction to the XND Ruby wrapper.

### CODE REVIEW

File: posts/2019/09/ruby-wrappers-for-the-xnd-project.md [hunk 1]
```diff
@@ -0,0 +1,388 @@
+<!--
+.. title: Ruby wrappers for the XND project
+.. slug: ruby-wrappers-for-the-xnd-project
+.. date: 2019-09-15 00:32:00 UTC-05:00
```
Comment (rgommers) [inline posts/2019/09/ruby-wrappers-for-the-xnd-project.md]: Can you also add your own name here with `.. author:` ?

File: posts/2019/09/ruby-wrappers-for-the-xnd-project.md [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+The available data types are limited and are extend to more complex uses. Modern use cases like
+Machine Learning require much more robust data types, as has been demonstrated by the tensor
+implementations of various frameworks like Tensorflow and PyTorch.
+
+The second problem is due to the fact that all of the aforementioned projects are community
+efforts that are maintained part-time by developers simply out of a sense of purpose and
+passion. Sustaining such complex projects for extended periods of time without expectation
+of any support is simply unfeasible even for the most driven engineers.
+
+This is where the XND project comes in. The [XND project](https://xnd.io/) is a project for
```
Comment (rgommers) [inline posts/2019/09/ruby-wrappers-for-the-xnd-project.md]: Can you add `<!-- TEASER_END -->` somewhere around here? (see README in this repo for what it does)

File: posts/2019/09/ruby-wrappers-for-the-xnd-project.md [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+of various types. All of them are able to easily interface with libraries like ATLAS, FFTW
+and LAPACK.
+
+However, all of the above projects fall short in two major aspects:
+* Lack of extensibility to adapt to modern use cases (read Machine Learning).
+* Lack of a critical mass of developers to maintain a robust and fast array library.
+
+The first problem is mainly due to the fact that they do not support very robust type systems.
+The available data types are limited and are extend to more complex uses. Modern use cases like
+Machine Learning require much more robust data types, as has been demonstrated by the tensor
```
Comment (rgommers) [inline posts/2019/09/ruby-wrappers-for-the-xnd-project.md]: "more robust data types" --> maybe " a more robust type system" or "more data types (such as `bfloat16`)"? I'm still a little confused by what you mean by this sentence

File: posts/2019/09/ruby-wrappers-for-the-xnd-project.md [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+These projects attempt to provide a simple Ruby-like API for creating and manipulating arrays
+of various types. All of them are able to easily interface with libraries like ATLAS, FFTW
+and LAPACK.
+
+However, all of the above projects fall short in two major aspects:
+* Lack of extensibility to adapt to modern use cases (read Machine Learning).
+* Lack of a critical mass of developers to maintain a robust and fast array library.
+
+The first problem is mainly due to the fact that they do not support very robust type systems.
+The available data types are limited and are extend to more complex uses. Modern use cases like
```
Comment (rgommers) [inline posts/2019/09/ruby-wrappers-for-the-xnd-project.md]: type: "are extend " --> "are hard to extend"

File: posts/2019/09/ruby-wrappers-for-the-xnd-project.md (hunk 3 above)
Comment (v0dro) [reply posts/2019/09/ruby-wrappers-for-the-xnd-project.md]: Like being able to define array shapes of arbitrary dimensions on multiple devices with a nice interface.

**Comment by v0dro** [comment]
@rgommers I've made the changes as you suggested.

File: posts/2019/09/ruby-wrappers-for-the-xnd-project.md [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+
+The XND project presents the perfect answer to Ruby's lack of a mature array computation ecosystem. 
+It is highly extensible, allows defining data types in almost any combination with a simple and
+intuitive interface, is built with performance in mind and is backed by a team consisting of 
+experts who have vast experience in this domain for the Python scientific computing stack.
+
+The biggest backer of XND as of now is Quansight, and I as a part-time engineer am responsible 
+for maintaining the Ruby wrapper for XND. This post is a rather long and detailed introduction 
+to the XND ruby wrapper including various use cases and benchmarks. There will also be some
+details on the implementation of the wrapper and how it differs from the python wrapper (which
```
Comment (rgommers) [inline posts/2019/09/ruby-wrappers-for-the-xnd-project.md]: minor: `python` --> `Python`

File: posts/2019/09/ruby-wrappers-for-the-xnd-project.md [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+elaborate on each of these in the post below.
+
+The XND project presents the perfect answer to Ruby's lack of a mature array computation ecosystem. 
+It is highly extensible, allows defining data types in almost any combination with a simple and
+intuitive interface, is built with performance in mind and is backed by a team consisting of 
+experts who have vast experience in this domain for the Python scientific computing stack.
+
+The biggest backer of XND as of now is Quansight, and I as a part-time engineer am responsible 
+for maintaining the Ruby wrapper for XND. This post is a rather long and detailed introduction 
+to the XND ruby wrapper including various use cases and benchmarks. There will also be some
```
Comment (rgommers) [inline posts/2019/09/ruby-wrappers-for-the-xnd-project.md]: "use cases and benchmarks" aren't actually discussed in the post currently

**Comment by v0dro** [comment]
@rgommers updated as per your feedback.

**Comment by rgommers** [comment]
Merged. Thanks @v0dro, very nice post!

**Pull request merged**